native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/common/states/Http2StateUtil.java

```diff
@@ -215,9 +216,12 @@ public static void writeHttp2Promise(Http2PushPromise pushPromise, ChannelHandle
         // Construct http request
         HttpRequest httpRequest = pushPromise.getHttpRequest();
         httpRequest.headers().add(HttpConversionUtil.ExtensionHeaderNames.SCHEME.text(), HTTP_SCHEME);
+        if (!httpRequest.headers().contains(HttpHeaderNames.HOST)) {
+            String host = ((InetSocketAddress) inboundRequestMsg.getProperties().get(LOCAL_ADDRESS)).getHostName();
```

Comment:
Calling the InetSocketAddress.getHostName() method can potentially incur a cost due to a reverse DNS lookup.
Can we use the InetSocketAddress.getAddress() method instead, which returns the IP address? would it be enough to satisfy the client?

Reply:
Updated to use `getAddress` https://github.com/ballerina-platform/module-ballerina-http/pull/1455/commits/11ee451e896de4c40b6bb0d750a44cdf954c7e32